torch/_inductor/codegen/common.py

```diff
@@ -1500,36 +1528,43 @@ def do_cse(v):
 
             @staticmethod
             def indirect_indexing(
-                var: CSEVariable, size: sympy.Expr, check: bool = True
+                var: CSEVariable, size: Union[sympy.Expr, int], check: bool = True
             ):
+                if isinstance(size, int):
+                    size = sympy.Integer(size)
+                assert isinstance(size, sympy.Expr), size
                 # Skip CSE since this doesn't return an expression
 
                 if var.bounds.lower < 0:  # type: ignore[operator]
+                    stm = ops.add(var, ops.index_expr(size, torch.long))
+                    # Mixed negative and non-negative
+                    if var.bounds.upper >= 0:  # type: ignore[operator]
+                        lt = ops.lt(var, 0)
+                        stm = ops.where(lt, stm, var)
```

Comment:
Why reorder this?

Reply:
I just figured it makes more readable to first have what the function really does, and after that the bound computation, which is much less important.